corehq/tests/test_locks.py

```diff
@@ -51,7 +52,7 @@ def simulate_reentrant_lock():
 @timelimit(0.1)
 def test_unreleased_lock():
     msg = "unreleased dict_values([ReentrantTestLock(name='unreleased', level=1)])"
-    with assert_raises(AssertionError, msg=msg):
+    with assert_raises(AssertionError, msg=re.compile("^" + re.escape(msg))):
```

Comment:
Is there a reason why `msg` can now be longer, or does this fix something that is currently wrong?

Reply:
This became necessary because pytest changes the assertion error message, I think as part of its assertion introspection logic:

```
>       assert not locks, f"unreleased {locks.values()}"
E       AssertionError: unreleased dict_values([ReentrantTestLock(name='unreleased', level=1)])
E       assert not {'unreleased': ReentrantTestLock(name='unreleased', level=1)}

corehq/tests/locks.py:58: AssertionError
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
... snip

>               self.gen.throw(typ, value, traceback)
E               AssertionError: not equal
E               "unreleased dict_values([ReentrantTestLock(name='unreleased', level=1)])\nassert not {'unreleased': ReentrantTestLock(name='unreleased', level=1)}"
E               "unreleased dict_values([ReentrantTestLock(name='unreleased', level=1)])"

../../.pyenv/versions/3.9.17/lib/python3.9/contextlib.py:137: AssertionError
```